compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -439,6 +439,13 @@ fn codegen_regular_intrinsic_call<'tcx>(
 
             ret.write_cvalue(fx, a);
         }
+        sym::is_val_statically_known => {
+            intrinsic_args!(fx, args => (_a); intrinsic);
+
+            // FIXME actually return `true` in case the argument is a constant
+            let res = fx.bcx.ins().iconst(cranelift_codegen::ir::types::I8, false as i64);
```

Comment:
You can use 0 instead of `false as i64` here. For the FIXME, it doesn't really make sense for Cranelift, Cranelift itself doesn't do inlining so it wouldn't add an instruction like this and when doing MIR inlining it should be possible to optimize the intrinsic call to a true at the MIR level.

Reply:
I used `false as i64` hoping it would be more readable than `0`, since the end goal is to return false. Do you think it didn’t have that effect?

I don’t know the details of how Cranelift works, but if it *does* run constant prolongation, it could make sense to return true after the function was inlined in a MIR pass. I do like your idea for MIR optimizations better though.

Reply:
The reason MIR optimization doesn't already resolve the intrinsic to true is because the frontend can't do everything the backend can (e.g. propagate floating point constants). When two crates have different optimization settings, it is possible for MIR optimizations to be run on a function but never backend optimizations. There is probably a good solution to this, but I think that is better left for another PR.

Reply:
Cranelift does have constprop, but I highly doubt they will ever add an instruction whose behavior depends on optimizations like this.

Reply:
> I used false as i64 hoping it would be more readable than 0, since the end goal is to return false. Do you think it didn’t have that effect?

I have been using 0 as false and 1 as true everywhere. I guess it doesn't matter much.

Reply:
I added the suggested changes in when I squashed.